src/main/java/org/osiam/resources/scim/Address.java

```diff
@@ -0,0 +1,397 @@
+                    @JsonProperty("streetAddress") String streetAddress,
+                    @JsonProperty("locality") String locality,
+                    @JsonProperty("region") String region,
+                    @JsonProperty("postalCode") String postalCode,
+                    @JsonProperty("country") String country,
+                    @JsonProperty("type") Type type) {
+        super(operation, value, display, primary, reference);
+        this.formatted = formatted;
+        this.streetAddress = streetAddress;
+        this.locality = locality;
+        this.region = region;
+        this.postalCode = postalCode;
+        this.country = country;
+        this.type = type;
+    }
+
+    /**
+     * Constructor for serialization. It is not intended for public use.
+     */
+    private Address(Builder builder) {
+        super(builder);
+        this.formatted = builder.formatted;
+        this.streetAddress = builder.streetAddress;
+        this.locality = builder.locality;
+        this.region = builder.region;
+        this.postalCode = builder.postalCode;
+        this.country = builder.country;
+        this.type = builder.type;
+    }
+
+    /**
+     * Gets the full mailing address, formatted for display or use with a mailing label.
+     *
+     * @return the formatted address
+     */
+    public String getFormatted() {
+        return formatted;
+    }
+
+    /**
+     * Gets the full street address, which may include house number, street name, etc.
+     *
+     * @return the street address
+     */
+    public String getStreetAddress() {
+        return streetAddress;
+    }
+
+    /**
+     * Gets the city or locality
+     *
+     * @return the city or locality
+     */
+    public String getLocality() {
+        return locality;
+    }
+
+    /**
+     * Gets the state or region
+     *
+     * @return region the region
+     */
+    public String getRegion() {
+        return region;
+    }
+
+    /**
+     * Gets the postal code
+     *
+     * @return postalCode the postal code
+     */
+    public String getPostalCode() {
+        return postalCode;
+    }
+
+    /**
+     * Gets the country name in ISO 3166-1 alpha 2 format, e.g. "DE" or "US".
+     *
+     * @return the country
+     */
+    public String getCountry() {
+        return country;
+    }
+
+    /**
+     * Gets the type of the attribute.
+     * <p/>
+     * <p>
+     * For more detailed information please look at the <a href=
+     * "http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-3.2" >SCIM core schema 2.0, section 3.2</a>
+     * </p>
+     *
+     * @return the actual type
+     */
+    public Type getType() {
+        return type;
+    }
+
+    @Override
+    public String getOperation() {
+        return super.getOperation();
+    }
+
+    @Override
+    public boolean isPrimary() {
+        return super.isPrimary();
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method hashCode() has an NPath complexity of 78125](https://www.codacy.com/app/OSIAM/connector4java/file/1980681097/issues/source?bid=3081107&fileBranchId=3117155#l171)